getting-started/setup-building.rst

```diff
@@ -655,3 +655,52 @@ every rule.
 
 
 .. _issue tracker: https://github.com/python/cpython/issues
+
+
+.. _using-codespaces:
+
+Contribute Using Github Codespaces
+===================================
+
+.. _codespaces-whats-codespaces:
+
+What is Github Codespaces
+-------------------------
+
+If you'd like to start contributing to CPython without needing to set up a local developer environment, you can use `Github Codespaces <https://github.com/features/codespaces>`_.
+Codespaces is a cloud-based development environment offered by GitHub that allows developers to write, build, test, and debug code directly within their web browser or in Visual Studio Code (VS Code).
+
+To help you get started CPython contains a `devcontainer folder <https://github.com/python/cpython/tree/main/.devcontainer>`_ with a json file that provides consistent and versioned codespace configurations for all users of the project.
+It also contains a docker file that allows you to set up the same environment but locally in a docker container if you'd prefer to do that.
+
+.. _codespaces-create-a-codespace:
+
+Create A CPython Codespace
+--------------------------
+
+Here are the basic steps needed to contribute a patch using Codespaces.
+You first need to navigate to the `CPython repo <https://github.com/python/cpython>`_ hosted on GitHub.
+
+Then you will need to:
+
+1. Click the green :guilabel:`Code` button and choose the ``codespaces`` tab.
+2. Press the green :guilabel:`create a new codespace on main` button.
+3. A screen should appear that lets you know your codespace is being set up. (Note: Since the CPython devcontainer is provided codespaces will use the configurations it specifies.)
+4. VS Code will open inside of your web browser, already linked up with your code and a terminal to the remote codespace.
+5. Use the terminal with the usual git commands to create a new branch, commit and push your changes once you're ready!
```

Comment:
```suggestion
3. A screen should appear that lets you know your codespace is being set up. (Note: Since the CPython devcontainer is provided, codespaces will use the configuration it specifies.)
4. VS Code will open inside your web browser, already linked up with your code and a terminal to the remote codespace.
5. Use the terminal with the usual Git commands to create a new branch, commit and push your changes once you're ready!
```